main.c

```diff
@@ -488,12 +669,21 @@ int main(int argc, char *argv[]) {
 	if (state.output_manager == NULL) {
 		fprintf(stderr, "compositor doesn't support "
 			"wlr-output-management-unstable-v1\n");
-		return EXIT_FAILURE;
+		goto error;
 	}
 
-	while (state.running && wl_display_dispatch(display) != -1) {
-		// This space intentionally left blank
+	int ret = loop(&state);
+	goto done;
```

Comment:
This goto-logic is complicated. Can we simplify it?

Reply:
Removed the `error` label, now there is only `done`.